build/net/src/lib.rs

```diff
@@ -82,23 +98,129 @@ pub fn load_net_config() -> Result<NetConfig> {
     Ok(cfg)
 }
 
-pub fn generate_vlan_consts(
+pub fn generate_port_consts(
     config: &NetConfig,
     mut out: impl std::io::Write,
 ) -> Result<(), std::io::Error> {
-    let vlan = config.vlan.unwrap();
-    let end = vlan.start + vlan.count;
-    if end > 0xFFF {
-        panic!("Invalid VLAN range (must be < 4096)");
+    let ports = config
+        .vlans
+        .values()
+        .map(|v| v.port)
+        .collect::<BTreeSet<_>>()
+        .len();
+    assert!(ports <= 2);
+    let port_count = ports.max(1);
+    let s = quote! {
+        pub const PORT_COUNT: usize = #port_count;
+    };
+    writeln!(out, "{s}")
+}
+
+fn check_vlan_config(config: &NetConfig) {
+    if config.vlans.values().any(|v| v.vid > 0xFFF) {
+        panic!("Invalid VLAN VID (must be < 4096)");
```

Comment:
nitpicky: maybe it would be better to use `Iterator::find` to actually return the first VID that's greater than 4096, so we can print what the value was?
```suggestion
    if let Some(v) = config.vlans.values().find(|v| v.vid > 0xFFF) {
        panic!("Invalid VLAN VID {} (must be < 4096)", v.vid);
```

Reply:
Sure, done